include/counter.h

```diff
@@ -27,30 +27,71 @@
 extern "C" {
 #endif
 
-typedef void (*counter_callback_t)(struct device *dev, void *user_data);
+/** @brief Single shot callback
+ *
+ * @param dev       Pointer to the device structure for the driver instance.
+ * @param id        Single shot request id.
+ * @param counter   Counter value that triggered the callback.
+ * @param user_data User data.
+ */
+typedef void (*counter_single_shot_callback_t)(struct device *dev,
+					       u32_t id,
+					       u32_t counter,
+					       void *user_data);
+
+struct counter_single_shot {
+	/** Index of the request. Maximal index is implementation dependent.*/
+	u8_t id;
+	/** If true ticks are treated as absolute value, else it is relative to
+	 * the counter reading performed during the call.
+	 */
+	bool abs;
+	u32_t ticks;
+	counter_single_shot_callback_t callback;
+	void *user_data;	
+};
+
+/* backward compatibility */
+#define counter_set_alarm counter_set_overflow
+
+typedef void (*counter_callback_t)(struct device *dev,
+					    void *user_data);
 
 typedef int (*counter_api_start)(struct device *dev);
 typedef int (*counter_api_stop)(struct device *dev);
 typedef u32_t (*counter_api_read)(struct device *dev);
-typedef int (*counter_api_set_alarm)(struct device *dev,
-				     counter_callback_t callback,
-				     u32_t count, void *user_data);
+typedef int (*counter_api_set_single_shot)(struct device *dev,
+					   struct counter_single_shot *params);
+
+typedef int (*counter_api_set_overflow)(struct device *dev,
+					u32_t ticks,
+					counter_callback_t callback,
+					void *user_data);
+
 typedef u32_t (*counter_api_get_pending_int)(struct device *dev);
+typedef int (*counter_api_ms_to_ticks)(struct device *dev,
```

Comment:
Do we really need separate us/ms calls? 
Could we just use us as a unit?
What about tick to ms/us conversion?

Reply:
us_to_ticks is enough indeed.

What would be the use case for the reverse function?

Reply:
us_to_ticks might not be enough. Consider the case when you have RTC prescaled to have 1hz freq and you want to get number of ticks for 12h timeout. 12h (12*3600*1000000 us) needs 36 bits to represent. Unless we use u64_t for us argument.

Reply:
is there such use case of counting for so many hours? (if so, then yes u64_t it would be)

Reply:
I see a use case when you would like to put device to very long sleep (e.g. 2 days). Normally, you could use system timer, but it operates on system clock (e.g. 32 khz). 2 days on 32k does not fit in 32 bits. Then you can take counter.h implementation on RTC, set frequency to 1hz and then you can set alarms for much longer periods (more than a month).

Reply:
In my opinion we should use 64-bit value with us as a unit. This gives us the most flexibility,